xapian-core/api/query.cc

```diff
@@ -92,10 +94,16 @@ Query::Query(op op_, Xapian::valueno slot, const std::string & limit)
 	    internal = new Xapian::Internal::QueryTerm();
 	else
 	    internal = new Xapian::Internal::QueryValueGE(slot, limit);
+    } else if (usual(op_ == OP_VALUE_GT)) {
+	internal = new Xapian::Internal::QueryValueGT(slot, limit);
     } else if (usual(op_ == OP_VALUE_LE)) {
 	internal = new Xapian::Internal::QueryValueLE(slot, limit);
+    } else if (usual(op_ == OP_VALUE_LT)) {
+	internal = new Xapian::Internal::QueryValueLT(slot, limit);
```

Comment:
The `usual(...)` macro here hints to the compiler that each of these branches is usually taken, which isn't correct.

Before the change it was, since supplying an invalid `op` is not the usual case.

So we shouldn't annotate these conditions now (except perhaps for the last one, but I did some experiments which suggest the compiler assumes that a branch which raises an exception shouldn't be predicted as taken, so it's probably superfluous to hint that it isn't explicitly).